public/app/features/alerting/unified/hooks/rules/usePendingPeriod.tsx

```diff
@@ -0,0 +1,25 @@
+import { useMemo } from 'react';
+
+import { CombinedRule } from 'app/types/unified-alerting';
+
+import { isRecordingRulerRule, isRecordingRule, isAlertingRulerRule, isAlertingRule } from '../../utils/rules';
+import { formatPrometheusDuration } from '../../utils/time';
+
+export function usePendingPeriod(rule: CombinedRule): string | undefined {
+  return useMemo(() => {
+    if (isRecordingRulerRule(rule.rulerRule) || isRecordingRule(rule.promRule)) {
```

Comment:
I would add a comment explaining why we are checking the pending period first in the ruler rule, and if it's not present , in the prom rule